viamonvif/device/device.go

```diff
@@ -87,6 +92,49 @@ type GetCapabilities struct {
 	Category onvif.CapabilityCategory `xml:"tds:Category"`
 }
 
+func isLocalIPAddress(hostname string) bool {
+	hostname = strings.TrimSpace(hostname)
+
+	if hostname == "localhost" {
+		return true
+	}
+
+	ip := net.ParseIP(hostname)
+	if ip == nil {
+		return false
+	}
+
+	if ip.IsLoopback() {
+		return true
+	}
+
+	// IPv4 private addresses.
```

Comment:
I think you just want [this method](https://pkg.go.dev/net/netip#Addr.IsPrivate) from the Go standard library

Reply:
Thanks